src/pylint_sonarjson/sonarjson_reporter.py

```diff
@@ -45,12 +45,13 @@ def _msg_to_sonar_dict(self, msg: Message):
                 "filePath": msg.path,
                 "textRange": {
                     "startLine": msg.line,
-                    "startColumn": msg.column,
                 }
             },
             "severity": self.sonar_checker.severity(msg),
             "effortMinutes": self.sonar_checker.effort(msg)
         }
+        if hasattr(msg, "column") and msg.column > 0:
```

Comment:
There's no need to check for the `column` attribute, it's always there -- those other checks (for e.g. `end_line`) were for compatibility with older pylint versions.

Besides, there's still a chance for a SonarQube error when `column` is bigger than 0 but has the same value as `end_column`. Hence it would be better to omit `end_column` ([which is optional in PyLint](https://github.com/PyCQA/pylint/blob/main/pylint/message/message.py#L32), whereas `column` is not) if `end_column` and `end_line` are the same as `column` and `line`, respectively.